packages/next/src/experimental/testmode/playwright/with-next.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  defineConfig,
+  devices,
+  type PlaywrightTestConfig,
+} from '@playwright/test'
+import { deepMerge } from '../../../lib/deep-merge'
+type withNextConfigFunction = (
+  config: PlaywrightTestConfig
+) => PlaywrightTestConfig
+
+/**
+ * A utility function for efficiently extending the Next.js default Playwright config
+ * @param configOrFunction
+ * @returns
+ */
+export function withNext(
+  configOrFunction: PlaywrightTestConfig | withNextConfigFunction = {}
+): PlaywrightTestConfig {
+  const config =
+    typeof configOrFunction === 'function'
+      ? configOrFunction(defaultPlaywrightConfig)
+      : deepMerge(defaultPlaywrightConfig, configOrFunction)
+  return defineConfig(config)
+}
+
+/**
+ * This is the default configuration generated by Playwright as of v1.43.0 with some modifications.
+ *
+ * - the `testMatch` property is configured to match all `*.spec.js` or `*.spec.ts` files within the `app` and `pages` directories
+ * - the `use` property is configured with a baseURL matching the expected dev server endpoint (http://127.0.0.1:3000)
+ * - the `webserver` property is configured to run `next dev`.
+ */
+export const defaultPlaywrightConfig: PlaywrightTestConfig = {
+  testMatch: '{app,pages}/**/*.spec.{t,j}s',
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
+  /* Opt out of parallel tests on CI. */
+  workers: process.env.CI ? 1 : undefined,
+  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
+  reporter: [['list'], ['html', { open: 'never' }]],
+  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+  use: {
+    /* Base URL to use in actions like `await page.goto('/')`. */
+    baseURL: 'http://127.0.0.1:3000',
```

Comment:
Does `next.config` allow setting port? In either case, it'd be good to have this customizable very easy since this is a very common thing to change. But maybe config merge is good enough.

Reply:
No it doesn't. This is something we'd have to add later, and I agree we need to be shareable